hypervisor/hw/pci.c

```diff
@@ -177,23 +177,27 @@ static inline uint32_t mmcfg_off_to_address(union pci_bdf bdf, uint32_t offset)
  */
 static uint32_t pci_mmcfg_read_cfg(union pci_bdf bdf, uint32_t offset, uint32_t bytes)
 {
-	uint32_t addr = mmcfg_off_to_address(bdf, offset);
-	void *hva = hpa2hva(addr);
-	uint32_t val;
-
+	uint32_t addr;
+	void *hva;
+	uint32_t val = 0U;
 
-	ASSERT(pci_is_valid_access(offset, bytes), "the offset should be aligned with 2/4 byte\n");
+	if (pci_is_valid_access(offset, bytes)) {
+		addr = mmcfg_off_to_address(bdf, offset);
+		hva = hpa2hva(addr);
 
-	switch (bytes) {
-	case 1U:
-		val = (uint32_t)mmio_read8(hva);
-		break;
-	case 2U:
-		val = (uint32_t)mmio_read16(hva);
-		break;
-	default:
-		val = mmio_read32(hva);
-		break;
+		switch (bytes) {
+		case 1U:
+			val = (uint32_t)mmio_read8(hva);
+			break;
+		case 2U:
+			val = (uint32_t)mmio_read16(hva);
+			break;
+		default:
+			val = mmio_read32(hva);
+			break;
+		}
+	} else {
+		pr_err("the offset should be aligned with 2/4 byte\n");
```

Comment:
If this error really happens, pr_err is insufficient.  We need to let the guest/VCPU die rather than the HV/PCPU die.